### ISSUE TITLE
[Issue]: HIP Windows (gfx1151) hard crash in H2D memcpy via torch.Tensor.to("cuda") with PyTorch 2.8 (ROCm 6.4.4 preview)

### ISSUE THREAD

**Issue by ajc9988** [ROCm-Developer-Tools/HIP#3874]
## Summary
On Windows 11 with AMD ROCm PyTorch preview 6.4.4, any host→device (H2D) copy causes a hard access violation inside amdhip64_6.dll on a Ryzen AI Max+ 395 / Radeon 8060S (gfx1151). `torch.cuda.init()` and device allocations succeed, but `Tensor.to("cuda")` or `CPU->GPU` copies crash in `c10::hip::memcpy_and_sync`.

## Environment
- OS: Windows 11 Pro (10.0.26100 64-bit)
- GPU: AMD Radeon(TM) 8060S Graphics (gfx1151)
- CPU: AMD RYZEN AI MAX+ 395 w/ Radeon 8060S
- Driver: AMD Software: PyTorch on Windows Preview Edition 25.20.01.14 (GUI shows 25.20.01.14; WMI shows 32.0.22001.14007)
- PyTorch: 2.8.0a0+gitfc14c65 (Windows ROCm preview)
- ROCm/HIP runtime: 6.4.50101-9a6572ae7 (HIP runtime 6.4.50101)
- MIOpen runtime: 3.4.1
- Python: 3.12.11 (conda-forge)
- Wheels installed FROM AMD repo:
  - torch:     https://repo.radeon.com/rocm/windows/rocm-rel-6.4.4/torch-2.8.0a0%2Bgitfc14c65-cp312-cp312-win_amd64.whl
  - torchaudio:https://repo.radeon.com/rocm/windows/rocm-rel-6.4.4/torchaudio-2.6.0a0%2B1a8f621-cp312-cp312-win_amd64.whl
  - torchvision:https://repo.radeon.com/rocm/windows/rocm-rel-6.4.4/torchvision-0.24.0a0%2Bc85f008-cp312-cp312-win_amd64.whl

`torch.utils.collect_env` excerpt:
- PyTorch version: 2.8.0a0+gitfc14c65
- ROCM used to build PyTorch: 6.4.50101-9a6572ae7
- Is CUDA available: True
- GPU model: AMD Radeon(TM) 8060S (gfx1151)
- HIP runtime version: 6.4.50101
- MIOpen runtime version: 3.4.1

## Minimal Repro (no RNG, isolates H2D/D2H)
```python
import torch
print("A:", torch.cuda.is_available())             # True
torch.cuda.init(); print("B: init ok")              # OK
x = torch.empty(1024, device="cuda"); print("C: empty cuda ok")  # OK

# E: device->host copy (often triggers)
xh = x.to("cpu"); torch.cuda.synchronize(); print("E: D2H ok")

# F: host->device copy (also triggers)
xh2 = torch.empty_like(xh)    # CPU
x2  = xh2.to("cuda")          # <-- hard crash in amdhip64_6.dll
torch.cuda.synchronize()
print("F: H2D ok")

Actual behavior
Process exits with access violation in amdhip64_6.dll during H2D/D2H path:

Exception Code: 0xC0000005
#0  amdhip64_6.dll+0x3a0ec7
#1  amdhip64_6.dll+0x227c3e
#2  amdhip64_6.dll+0x207268
#3  c10_hip.dll! c10::hip::memcpy_and_sync(void*, const void*, __int64, hipMemcpyKind, ihipStream_t*)
...
#5  torch_cpu.dll! at::native::copy_(...)
#8  torch_cpu.dll! at::native::_to_copy(...)
#13 torch_cpu.dll! at::_ops::_to_copy::call(...)
#17 torch_cpu.dll! at::Tensor::to(...)
#18 torch_python.dll ...


Expected behavior
Tensor.to("cuda") should perform a normal H2D copy without crashing.
Also observed
Pure device allocation succeeds: torch.empty(..., device="cuda") works.
RNG on GPU also crashes (separate trace in torch_hip.dll via at::CUDAGeneratorImpl::reset_rnn_state), but even without RNG the H2D path alone triggers the crash.
What I tried
Verified wheels from AMD repo (not generic pip).
Verified driver is the AMD PyTorch Preview 25.20.01.14.
Unset NVIDIA var: CUDA_MODULE_LOADING.
Tried HIP debug flags: AMD_SERIALIZE_KERNEL=3, TORCH_USE_HIP_DSA=1.
Disabled SDMA: HSA_ENABLE_SDMA=0 (no change).
Removed cross backends from env: torch-directml, nvidia-cuda-runtime-*.
Rebooted and re-tested in a clean shell. Same crash at H2D/D2H.

python -m torch.utils.collect_env
<frozen runpy>:128: RuntimeWarning: 'torch.utils.collect_env' found in sys.modules after import of package 'torch.utils', but prior to execution of 'torch.utils.collect_env'; this may result in unpredictable behaviour
Collecting environment information...
PyTorch version: 2.8.0a0+gitfc14c65
Is debug build: False
CUDA used to build PyTorch: N/A
ROCM used to build PyTorch: 6.4.50101-9a6572ae7

OS: Microsoft Windows 11 Pro (10.0.26100 64-bit)
GCC version: (conda-forge gcc 15.2.0-7) 15.2.0
Clang version: 21.1.2
CMake version: version 4.1.2
Libc version: N/A

Python version: 3.12.11 | packaged by conda-forge | (main, Jun  4 2025, 14:29:09) [MSC v.1943 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-11-10.0.26100-SP0
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: AMD Radeon(TM) 8060S Graphics (gfx1151)
Nvidia driver version: Could not collect
cuDNN version: Could not collect
HIP runtime version: 6.4.50101
MIOpen runtime version: 3.4.1
Is XNNPACK available: False

CPU:
Name: AMD RYZEN AI MAX+ 395 w/ Radeon 8060S
Manufacturer: AuthenticAMD
Family: 107
Architecture: 9
ProcessorType: 3
DeviceID: CPU0
CurrentClockSpeed: 3000
MaxClockSpeed: 3000
L2CacheSize: 16384
L2CacheSpeed: None
Revision: 28672

Versions of relevant libraries:
[pip3] adam-atan2-pytorch==0.2.2
[pip3] botorch==0.15.1
[pip3] fast_pytorch_kmeans==0.2.2
[pip3] flake8==7.3.0
[pip3] flake8-2020==1.8.1
[pip3] flake8-alfred==1.1.1
[pip3] flake8_all_not_strings==0.0.3
[pip3] flake8-annotations-complexity==0.1.0
[pip3] flake8-async==25.7.1
[pip3] flake8-bandit==4.1.1
[pip3] flake8-black==0.4.0
[pip3] flake8-bugbear==24.12.12
[pip3] flake8-builtins==3.0.0
[pip3] flake8-class-attributes-order==0.3.0
[pip3] flake8-clean-block==0.1.2
[pip3] flake8-comprehensions==3.17.0
[pip3] flake8-datetimez==20.10.0
[pip3] flake8-docstrings==1.7.0
[pip3] flake8-dunder-all==0.5.0
[pip3] flake8-eradicate==1.5.0
[pip3] flake8-for-pycharm==0.4.1
[pip3] flake8-helper==0.2.2
[pip3] flake8-html==0.4.3
[pip3] flake8-isort==6.1.2
[pip3] flake8-json==24.4.0
[pip3] flake8-logging-format==2024.24.12
[pip3] flake8-markdown==0.6.0
[pip3] flake8-mock-spec==1.4.0
[pip3] flake8-mutable==1.2.0
[pip3] flake8-pie==0.16.0
[pip3] flake8-plugin-utils==1.3.3
[pip3] flake8-polyfill==1.0.2
[pip3] flake8-pytest-style==2.1.0
[pip3] flake8-pytestrail==0.2.1
[pip3] flake8-qt-tr==0.2.0
[pip3] flake8-requirements==2.3.0
[pip3] flake8-scrapy==0.0.1
[pip3] flake8-secure-coding-standard==1.4.1
[pip3] flake8_simplify==0.22.0
[pip3] flake8-string-format==0.3.0
[pip3] flake8-tergeo==25.10.9.0
[pip3] flake8-tidy-imports==4.12.0
[pip3] flake8_typing_imports==1.16.0
[pip3] flake8-unused-arguments==0.0.13
[pip3] flake8-useless-assert==0.4.4
[pip3] flake8-warnings==0.4.1
[pip3] flamingo-pytorch==0.1.2
[pip3] gpytorch==1.14
[pip3] hypothesis-torch==2.0.5
[pip3] mypy==1.18.2
[pip3] mypy_extensions==1.1.0
[pip3] numpy==1.26.4
[pip3] numpyro==0.19.0
[pip3] onnx==1.18.0
[pip3] onnx-ir==0.1.10
[pip3] onnx-tool==0.9.0
[pip3] onnxconverter-common==1.16.0
[pip3] onnxoptimizer==0.3.19
[pip3] onnxruntime==1.23.1
[pip3] onnxruntime_extensions==0.14.0
[pip3] onnxruntime-genai-directml-ryzenai==0.9.2
[pip3] onnxruntime-vitisai==1.23.0
[pip3] onnxscript==0.5.3
[pip3] onnxsim==0.4.36
[pip3] open_clip_torch==3.2.0
[pip3] optimum-onnx==0.0.3
[pip3] optree==0.17.0
[pip3] pytest-mypy-plugins==3.2.0
[pip3] pytorch-lightning==2.5.2
[pip3] pytorch-metric-learning==2.9.0
[pip3] pytorch_revgrad==0.2.0
[pip3] ryzenai_onnx_utils==1.6.0
[pip3] torch==2.8.0a0+gitfc14c65
[pip3] torch-audiomentations==0.12.0
[pip3] torch-geometric==2.6.1
[pip3] torch-max-mem==0.1.4
[pip3] torch_pitch_shift==1.2.5
[pip3] torch-ppr==0.0.8
[pip3] torch_uncertainty==0.7.0.post1
[pip3] torchaudio==2.6.0a0+1a8f621
[pip3] torchdata==0.11.0
[pip3] torchmetrics==1.8.2
[pip3] torchtune==0.6.1
[pip3] torchvision==0.24.0a0+c85f008
[pip3] triton==3.0.0
[conda] adam-atan2-pytorch                                   0.2.2                  pypi_0              pypi
[conda] botorch                                              0.15.1                 pypi_0              pypi
[conda] fast-pytorch-kmeans                                  0.2.2                  pypi_0              pypi
[conda] flamingo-pytorch                                     0.1.2                  pypi_0              pypi
[conda] gpytorch                                             1.14                   pypi_0              pypi
[conda] hypothesis-torch                                     2.0.5                  pypi_0              pypi
[conda] numpy                                                1.26.4                 pypi_0              pypi
[conda] numpyro                                              0.19.0                 pypi_0              pypi
[conda] open-clip-torch                                      3.2.0                  pypi_0              pypi
[conda] optree                                               0.17.0                 pypi_0              pypi
[conda] pytorch-lightning                                    2.5.2                  pypi_0              pypi
[conda] pytorch-metric-learning                              2.9.0                  pypi_0              pypi
[conda] pytorch-revgrad                                      0.2.0                  pypi_0              pypi
[conda] torch                                                2.8.0a0+gitfc14c65     pypi_0              pypi
[conda] torch-audiomentations                                0.12.0                 pypi_0              pypi
[conda] torch-geometric                                      2.6.1                  pypi_0              pypi
[conda] torch-max-mem                                        0.1.4                  pypi_0              pypi
[conda] torch-pitch-shift                                    1.2.5                  pypi_0              pypi
[conda] torch-ppr                                            0.0.8                  pypi_0              pypi
[conda] torch-uncertainty                                    0.7.0.post1            pypi_0              pypi
[conda] torchaudio                                           2.6.0a0+1a8f621        pypi_0              pypi
[conda] torchdata                                            0.11.0                 pypi_0              pypi
[conda] torchmetrics                                         1.8.2                  pypi_0              pypi
[conda] torchtune                                            0.6.1                  pypi_0              pypi
[conda] torchvision                                          0.24.0a0+c85f008       pypi_0              pypi
[conda] triton                                               3.0.0                  pypi_0              pypi

Get-WmiObject Win32_VideoController `
>>   | Where-Object {$_.Name -like "*AMD*" -or $_.Name -like "*Radeon*"} `
>>   | Select-Object Name, DriverVersion

Name                          DriverVersion
----                          -------------
AMD Radeon(TM) 8060S Graphics 32.0.22001.14007

python -c "import sys; import os; sys.path.insert(0, os.path.join(os.environ['CONDA_PREFIX'], 'Lib', 'site-packages')); import flash_attn; print(flash_attn.__file__)"
C:\Users\*******\.conda\envs\********\Lib\site-packages\flash_attn\__init__.py

### Operating System

Windows 11 Pro

### CPU

AMD Ryzen AI Max+ 395

### GPU

8060s

### ROCm Version

ROCm 6.4.50101-9a6572ae7 (HIP SDK)

### ROCm Component

_No response_

### Steps to Reproduce

_No response_

### (Optional for Linux users) Output of /opt/rocm/bin/rocminfo --support

_No response_

### Additional Information

_No response_
